sqlalchemy_bigquery/_json.py

```diff
@@ -0,0 +1,8 @@
+import sqlalchemy
+
+
+class JSON(sqlalchemy.sql.sqltypes.JSON):
+    def bind_expression(self, bindvalue):
+        # JSON query parameters have type STRING
+        # This hook ensures that the rendered expression has type JSON
+        return sqlalchemy.func.PARSE_JSON(bindvalue, type_=self)
```

Comment:
Looking at [1] and pondering this - it wouldn't be that surprising if a user wanted to submit _invalid_ JSON as a query parameter at some point. This could be supported by parameterizing this behavior in the type, i.e: `JSON(strict=False)` or similar. Not worried about it right now, though.

[1] https://cloud.google.com/bigquery/docs/reference/standard-sql/json_functions#non-validation_of_string_inputs